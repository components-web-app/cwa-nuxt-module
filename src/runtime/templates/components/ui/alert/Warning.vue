<template>
  <div
    class="cwa:flex cwa:items-start cwa:outline-1 cwa:outline-offset-2 cwa:outline-red-500/50 cwa:bg-red-500/50 cwa:text-white cwa:text-sm cwa:font-bold cwa:px-4 cwa:py-3 cwa:rounded-md"
    role="alert"
  >
    <slot />
  </div>
</template>
