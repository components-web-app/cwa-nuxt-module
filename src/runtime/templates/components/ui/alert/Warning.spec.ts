// @vitest-environment happy-dom
import { describe, expect, test } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import Warning from './Warning.vue'

function createWrapper(content: string) {
  return shallowMount(Warning, {
    slots: {
      default: content,
    },
  })
}
describe('Warning', () => {
  describe('snapshots', () => {
    test('should match snapshot with passed content', () => {
      const wrapper = createWrapper('<div>CWA</div>')

      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
