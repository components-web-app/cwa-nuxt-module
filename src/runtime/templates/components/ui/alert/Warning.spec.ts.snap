// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Warning > snapshots > should match snapshot with passed content 1`] = `
<div
  class="cwa:flex cwa:items-start cwa:outline-1 cwa:outline-offset-2 cwa:outline-red-500/50 cwa:bg-red-500/50 cwa:text-white cwa:text-sm cwa:font-bold cwa:px-4 cwa:py-3 cwa:rounded-md"
  role="alert"
>
  
  <div>
    CWA
  </div>
  
</div>
`;
